src/pybind/mgr/rook/rook_cluster.py

```diff
@@ -408,14 +414,21 @@ def add_osds(self, drive_group, all_hosts):
             assert node_idx is not None
             assert current_node is not None
 
-            new_devices = list(set(block_devices) - set([d['name'] for d in current_node['devices']]))
+            new_devices = list(set(paths) - set([d['name'] for d in current_node['devices']]))
 
             for n in new_devices:
-                patch.append({
-                    "op": "add",
-                    "path": "/spec/storage/nodes/{0}/devices/-".format(node_idx),
-                    "value": {'name': n}
-                })
+                if drive_group.objectstore is "bluestore":
```

Comment:
It's possible to also have bluestore on a directory and filestore on a device, but we should keep it simple like this. Filestore is only expected for test scenarios anymore, right?

Reply:
Yes, I think so. If you want to do anything fancy like that, then you're probably better off updating the CephCluster object directly.

Reply:
After talking it over with @batrick , I think I'm going to convert this to just use bluestore on a directory, since the filestore code is heading toward deprecation. I'll rework this to do that (and do some testing with it).